network/dag/notifier.go

```diff
@@ -239,16 +239,31 @@ func (p *notifier) Run() error {
 	if !p.isPersistent() {
 		return nil
 	}
-	return p.db.ReadShelf(p.ctx, p.shelfName(), func(reader stoabs.Reader) error {
+	// we're going to retry all events synchronously at startup. For the ones that fail we'll start the retry loop
+	failedAtStartup := make([]Event, 0)
+	err := p.db.ReadShelf(p.ctx, p.shelfName(), func(reader stoabs.Reader) error {
 		return reader.Iterate(func(k stoabs.Key, v []byte) error {
 			event := Event{}
 			_ = json.Unmarshal(v, &event)
 
-			p.retry(event)
+			if err := p.notifyNow(event); err != nil {
+				failedAtStartup = append(failedAtStartup, event)
+			}
```

Comment:
this changes some logic. Events with `Event.Retries >= maxRetries` should not be retried.

Reply:
> this changes some logic. Events with `Event.Retries >= maxRetries` should not be retried.

That is already the case @ startup

Reply:
I think the checks are in the wrong order? It still retries when the maxRetries is reached. (might also fix the broken test)

Reply:
hmm... I see what you're saying. Everything is retried at startup after the initial delay, which is 2^100 sec == never for events that hit an `EventFatal` error. `p.retry` is basically collecting go routines that sleep forever...?

Reply:
time.Second^100 overflows and results in `0s` in go, so it retries immediately. I'd say that this is a different bug? 

Also, `retry.Attempts(maxRetries-uint(initialCount)),` in `p.retry` result in infinite retries when `initialCount== maxRetries`

mehhh

Reply:
Added Jitter as well. Hopefully the test isn't going to random fail